Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -5686,17 +5687,35 @@ - (void)setMarkedTextStyle:(NSDictionary *)styleDictionary
 #if USE(APPLE_INTERNAL_SDK)
 #include <WebKitAdditions/WKContentViewInteractionAdditions.mm>
 #else
-- (void)setAttributedMarkedText:(NSAttributedString *)markedText selectedRange:(NSRange)selectedRange
+- (void)setMarkedText:(NSString *)markedText selectedRange:(NSRange)selectedRange
 {
-    [self _setMarkedText:markedText.string underlines:Vector<WebCore::CompositionUnderline> { } highlights:compositionHighlights(markedText) selectedRange:selectedRange];
+    [self _setMarkedText:markedText underlines:Vector<WebCore::CompositionUnderline> { } highlights:Vector<WebCore::CompositionHighlight> { } selectedRange:selectedRange];
 }
 
-- (void)setMarkedText:(NSString *)markedText selectedRange:(NSRange)selectedRange
+static inline bool shouldExtractMarkedTextComposition()
 {
-    [self _setMarkedText:markedText underlines:Vector<WebCore::CompositionUnderline> { } highlights:Vector<WebCore::CompositionHighlight> { } selectedRange:selectedRange];
+    return false;
 }
 #endif
 
+- (void)setAttributedMarkedText:(NSAttributedString *)markedText selectedRange:(NSRange)selectedRange
+{
+    WebKit::MarkedTextComposition composition;
+    if (shouldExtractMarkedTextComposition())
+        composition = WebKit::extractMarkedTextComposition(markedText);
+    else
+        composition = compositionHighlights(markedText);
+
+    WTF::switchOn(composition, [&](std::monostate) {
+        [self _setMarkedText:markedText.string underlines:Vector<WebCore::CompositionUnderline> { } highlights:Vector<WebCore::CompositionHighlight> { } selectedRange:selectedRange];
```

Comment:
Nit - does it work if you replace the `Vector<WebCore::CompositionUnderline> { }` with just `{ }`? (If so, it might make this a bit cleaner)